### PR TITLE
Fix uncollect condition and delete bz blocker

### PR DESCRIPTION
**Pull request by lehinevych**
{{pytest: cfme/tests/infrastructure/test_vm_retirement_rest.py -v --long-running }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 55e8055c418fceb0420a8142e048e26bfc2b414a:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by lehinevych** [comment]
Is blocked by https://github.com/ManageIQ/manageiq/issues/7293

File: cfme/tests/infrastructure/test_vm_retirement_rest.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 import fauxfactory
 
 from cfme.rest import a_provider as _a_provider
-from utils.version import appliance_is_downstream
+from utils.version import current_version
```
Comment (mfalesni) [inline cfme/tests/infrastructure/test_vm_retirement_rest.py]: @psav was this the thing you were saying in the call? That we will get rid of this convenience function?

Comment (lehinevych) [inline cfme/tests/infrastructure/test_vm_retirement_rest.py]: @mfalesni I think yes and it should be replace to store.current_appliance.version

Comment (psav) [inline cfme/tests/infrastructure/test_vm_retirement_rest.py]: Absolutely!

**Pull request merged**